packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -131,12 +133,18 @@ export default class ScriptTransformer {
   }
 
   private _getTransformPath(filename: Config.Path) {
-    for (let i = 0; i < this._config.transform.length; i++) {
-      if (new RegExp(this._config.transform[i][0]).test(filename)) {
-        return this._config.transform[i][1];
+    const transformRegExp = this._cache.transformRegExp;
+    if (!transformRegExp) {
```

Comment:
When can this be falsy?

Reply:
When the array passed from config with transforms is 0-length.

Reply:
Ah, `transform: {}`? Gotcha

Reply:
`calcTransformRegExp` returns `undefined` if `config.transform.length === 0`.

I try not to return empty arrays to loop through when the function should actually be telling you "nothing to do here". I personally find it makes the code more readable and sometimes make optimizations in the future possible.